R/validate_target_data.R

```diff
@@ -0,0 +1,186 @@
+#'   dplyr::mutate("Extra info" = dplyr::case_when(
+#'     is.na(.data$`Extra info`) ~ "",
+#'     TRUE ~ .data$`Extra info`
+#'   )) %>%
+#'   knitr::kable() %>%
+#'   kableExtra::kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive")) %>%
+#'   kableExtra::column_spec(1, bold = TRUE)
+#' ```
+#' @examples
+#' hub_path <- system.file("testhubs/v5/target_file", package = "hubUtils")
+#' validate_target_data(hub_path,
+#'   file_path = "time-series.csv",
+#'   target_type = "time-series"
+#' )
+#' validate_target_data(hub_path,
+#'   file_path = "oracle-output.csv",
+#'   target_type = "oracle-output"
+#' )
+#' hub_path <- system.file("testhubs/v5/target_dir", package = "hubUtils")
+#' validate_target_data(hub_path,
+#'   file_path = "time-series/target=wk%20flu%20hosp%20rate/part-0.parquet",
+#'   target_type = "time-series"
+#' )
+#' validate_target_data(hub_path,
+#'   file_path = "oracle-output/output_type=pmf/part-0.parquet",
+#'   target_type = "oracle-output"
+#' )
+#' @export
+validate_target_data <- function(
+  hub_path,
+  file_path,
+  target_type = c("time-series", "oracle-output"),
+  date_col = NULL,
+  na = c("NA", ""),
+  output_type_id_datatype = c(
+    "from_config",
+    "auto",
+    "character",
+    "double",
+    "integer",
+    "logical",
+    "Date"
+  ),
+  validations_cfg_path = NULL,
+  round_id = "default"
+) {
+  target_type <- rlang::arg_match(target_type)
+  output_type_id_datatype <- rlang::arg_match(output_type_id_datatype)
+
+  checks <- new_hub_validations()
+
+  # -- File parsing checks ----
+  checks$target_file_read <- try_check(
+    check_target_file_read(
+      file_path = file_path,
+      hub_path = hub_path
+    ),
+    file_path
+  )
+  if (is_any_error(checks$target_file_read)) {
+    return(checks)
+  }
+
+  # If `csv` file, read in using hub schema. Otherwise use file
+  # schema for other file formats.
+  if (fs::path_ext(file_path) == "csv") {
+    target_tbl <- read_target_file(
+      target_file_path = file_path,
+      hub_path = hub_path,
+      coerce_types = "target"
+    )
+  } else {
+    target_tbl <- read_target_file(
+      target_file_path = file_path,
+      hub_path = hub_path,
+      coerce_types = "none"
+    )
+  }
+
+  # -- Standard checks ----
+  checks$target_tbl_colnames <- try_check(
+    check_target_tbl_colnames(
+      target_tbl = target_tbl,
+      target_type = target_type,
+      file_path = file_path,
+      hub_path = hub_path
+    ),
+    file_path
+  )
+  if (is_any_error(checks$target_tbl_colnames)) {
+    return(checks)
+  }
+  checks$target_tbl_coltypes <- try_check(
+    check_target_tbl_coltypes(
+      target_tbl = target_tbl,
+      target_type = target_type,
+      date_col = date_col,
+      na = na,
+      output_type_id_datatype = output_type_id_datatype,
+      file_path = file_path,
+      hub_path = hub_path
+    ),
+    file_path
+  )
+  if (is_any_error(checks$target_tbl_coltypes)) {
+    return(checks)
+  }
```

Comment:
Can I ask why this one stop the checking and returns an error? Will it cause issue in the next checks if it fails?

Reply:
The oracle value check assumes that the column is the correct data type so is affected. The ts_target check also assumes the target column is the correct data type. But more importantly column data type is an important property that custom checks would also assume has already been checked and is correct.